### ISSUE TITLE
Upgrading from 3.0.7 to 3.1.0 breaks JpaSpecificationExecutor queries for model with inheritance

### ISSUE THREAD

**Issue by bjornhilstad** [spring-projects/spring-data-jpa#2985]
Error produced is:
```
Request processing failed: java.lang.NullPointerException: Cannot invoke "org.hibernate.sql.ast.tree.from.TableGroup.findTableGroupJoin(org.hibernate.sql.ast.tree.from.TableGroup)" because "parentParentTableGroup" is null] with root cause 
java.lang.NullPointerException: Cannot invoke "org.hibernate.sql.ast.tree.from.TableGroup.findTableGroupJoin(org.hibernate.sql.ast.tree.from.TableGroup)" because "parentParentTableGroup" is null
```

A simple reproducer is submitted here.
Update build.gradle to control spring-boot version (3.0.7 and 3.1.0 tested).
3.1.0 can also work if hibernate is downgraded to 6.1.7.Final

API that triggers Specification based search for model that uses inheritance can be done with:
```
curl --location 'http://localhost:8080/api/open/address/search' \
--header 'Content-Type: application/json' \
--data '{
    "streetName": "Name"
}'
``` 
This produces the error for 3.1.0 but works with 3.0.7.

API that triggers Specification based search that does not use inheritance can de done with:
```
curl --location 'http://localhost:8080/api/open/street/search' \
--header 'Content-Type: application/json' \
--data '{
    "streetName": "BES"
}'
``` 
This call does not fail on either version.

[hibernate-bug.zip](https://github.com/spring-projects/spring-data-jpa/files/11576943/hibernate-bug.zip)

**Comment by mp911de** [comment]
With a deeply nested stacktrace in Hibernate code, what is the indicator that Spring Data is the the right place to file this report? Also if downgrading to an earlier Hibernate version remediates the issue, there's no indication that Spring Data can address your issue.

Please file the bug report against the Hibernate issue tracker.

**Comment by bjornhilstad** [comment]
I filed the issue here becaues I use hibernate through spring jpa so I thought the issue might be cause by some interaction between the two.
But I made a pure hibernate test that reproduces the issue and submitted this to the Hibernate issue tracker, https://hibernate.atlassian.net/browse/HHH-16719

**Comment by bjornhilstad** [comment]
Final update: The reported hibernate issue is now fixed in hibernate 6.2.5.Final
